src/data.ts

```diff
@@ -161,7 +161,7 @@ export function getFirst(query: any | string | RegExp) : Promise<IDataType> {
     return getFirstByName(<string>query);
   }
   query.limit = 1;
-  return list(query).then((result) => {
+  return list(query).then<IDataType>((result) => {
```

Comment:
@mccalluc This defines the result type of the promise. I don't know why the auto inference doesn't work in this case.

See also the `then()` definition:
```
    /**
     * Attaches callbacks for the resolution and/or rejection of the Promise.
     * @param onfulfilled The callback to execute when the Promise is resolved.
     * @returns A Promise for the completion of which ever callback is executed.
     */
    then<TResult>(onfulfilled: (value: T) => TResult | PromiseLike<TResult>): Promise<TResult>;
```